testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -171,16 +171,43 @@ class TestContacts(object):
     def universe():
         return mda.Universe(PSF, DCD)
 
-    def _run_Contacts(self, universe, start=None, stop=None, step=None, **kwargs):
+    def _run_Contacts(
+        self, universe,
+        start=None, stop=None, step=None, **kwargs
+    ):
         acidic = universe.select_atoms(self.sel_acidic)
         basic = universe.select_atoms(self.sel_basic)
+
         return contacts.Contacts(
             universe,
             select=(self.sel_acidic, self.sel_basic),
             refgroup=(acidic, basic),
             radius=6.0,
             **kwargs).run(start=start, stop=stop, step=step)
 
+    @pytest.mark.parametrize("seltxt", [sel_acidic, sel_basic])
+    def test_select_valid_types(self, universe, seltxt):
```

Comment:
Could you add one more test, almost the same as this one but where you actually `.run()` the contacts with a `string` and `AtomGroup` input and compare the results?


You have tested the internal `staticmethod` `Contacts.get_atomgroup()` but not the actual flow of data through the whole function when a `string ` vs `AtomGroup` is used.

Reply:
Done! See [line 199:](https://github.com/MDAnalysis/mdanalysis/pull/3565/files#diff-c79b8e25558e1ba2f042d25eaa0e355502afba57cf03bf8c9b22e0c6103063a4R199)
```python
def test_contacts_selections(self, universe):
```